pkg/server/metrics.go

```diff
@@ -0,0 +1,72 @@
+package server
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/google/go-github/github"
+	"github.com/arquillian/ike-prow-plugins/pkg/log"
+)
+
+var (
+	namespace = ""
+	subsystem = "ike-prow-plugins"
+
+	rateLimit = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Name: "github_rate_limits",
+		Help: "Rate limit for GitHub API calls",
+	}, []string{"type"})
+
+	webhookCounter = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Name: "webhook_counter",
+		Help: "A counter of the webhooks made to service.",
+	}, []string{"full_name"})
+
+	handledEventsCounter = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Name: "handled_events_counter",
+		Help: "A counter of handled events.",
+	}, []string{"event_type"})
+)
+
+// RegisterMetrics registers prometheus collectors to collect metrics
+func RegisterMetrics(log log.Logger) {
+	rateLimit = register(rateLimit, "github_rate_limits", log).(*prometheus.GaugeVec)
+	webhookCounter = register(webhookCounter, "webhook_counter", log).(*prometheus.CounterVec)
+	handledEventsCounter = register(handledEventsCounter, "handled_events_counter", log).(*prometheus.CounterVec)
+}
+
+func register(c prometheus.Collector, name string, log log.Logger) prometheus.Collector {
+	err := prometheus.Register(c)
+	if err != nil {
+		if are, ok := err.(prometheus.AlreadyRegisteredError); ok {
+			return are.ExistingCollector
+		}
+
+		log.Panic(map[string]interface{}{
```

Comment:
you have three options to avoid adding the `Panic` method to the interface:
1. return an array of errors that should be reported inside of the `plugin_init.go` using `logger.WithError(err).Fatalf`. if the array is empty, then everything is OK
2. instead of using our interface as a type of the `log` parameter in this function use the `*logrus.Entry` directly so you can fail using the same method  `logger.WithError(err).Fatalf` directly from this place (don't forget to accumulate the error)
3. keep the type of the parameter as it is (our interface), log every "register error" separately using `error` method. When there is some error return a new one (eg. `errors.New("Prometheus metrics registration failed. See preceding errors.")errors.New("Prometheus metrics registration failed. See preceding errors.")`) to the `plugin_init.go` where you can fail.